### DISCUSSION TITLE
Sample code snippet for utilizing observable state's options?

### DISCUSSION THREAD

**Discussion by theresa-de-ocampo** [mobxjs/mobx#3420] [Q&A] [answered]
I'm new to MobX, and was reading [https://mobx.js.org/observable-state.html#makeobservable](url). It provided examples for class + `makeObservable`, factory function + `makeAutoObservable`, and `observable`. But none of the code snippets showed an example of utlizing [https://mobx.js.org/observable-state.html#options-](url). Can anyone provide a sample code snippet for utilizing observable state's options? Even just for the the `name`, since it seems like it will be the most useful for beginners who needs to debug quite a lot.

P.S. It's also my first time posting on a GitHub discussion, let me know if I wasn't able to follow some standards.

**Comment by kubk** [comment, marked as answer]
Hi, here is an example utilizing `autoBind` option: https://codesandbox.io/s/minimal-mobx-react-project-ppgml?file=/index.js

For debugging you may find it useful: https://github.com/kubk/mobx-log
